### PR TITLE
[Docs] update Dynamic-compile document

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request

update the Dynamic-Compile transform document.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liunaijie** [comment]
@jackyyyyyssss PTAL

**Comment by Hisoka-X** [comment]
cc @jackyyyyyssss

**Review by jackyyyyyssss — APPROVED** [review]

**Comment by jackyyyyyssss** [comment]
LGTM. Thanks @liunaijie

File: docs/en/transform-v2/dynamic-compile.md [hunk 1]
```diff
@@ -47,6 +42,28 @@ If it is a SOURCE-CODE enumeration; the SOURCE-CODE attribute is required, and t
 
 The absolute path of Java or Groovy files on the server
 
+### source_code [string]
+
+The source code.
+
+### Detail about the Source Code
```
Comment (Hisoka-X) [inline docs/en/transform-v2/dynamic-compile.md]:
```suggestion
#### Details about the source code
```

File: docs/zh/transform-v2/dynamic-compile.md [hunk 2]
```diff
@@ -47,6 +42,26 @@ SOURCE_CODE,ABSOLUTE_PATH
 
 服务器上Java或Groovy文件的绝对路径
 
+### source_code [string]
+源代码
+
+#### 关于源代码
```
Comment (Hisoka-X) [inline docs/zh/transform-v2/dynamic-compile.md]:
```suggestion
#### 关于source_code
```

**Review by Hisoka-X — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**